ddl/rollingback.go

```diff
@@ -236,12 +236,31 @@ func rollingbackAddIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job, isP
 	return
 }
 
-func rollingbackAddTablePartition(t *meta.Meta, job *model.Job) (ver int64, err error) {
-	_, err = getTableInfoAndCancelFaultJob(t, job, job.SchemaID)
+func convertAddTablePartitionJob2RollbackJob(t *meta.Meta, job *model.Job, otherwiseErr error) (ver int64, err error) {
+	job.State = model.JobStateRollingback
+	tblInfo, _, addingDefinitions, err := checkAddPartition(t, job)
```

Comment:
When `convertAddTablePartitionJob2RollbackJob` is called in `onAddTablePartition`, I think we needn't to call `checkAddPartition `.

Reply:
great！addressed